// @vuepress/plugin-catalog@2.0.0-rc.47/lib/node/catalogPlugin.js
import { addViteSsrNoExternal, getLocaleConfig } from '@vuepress/helper';
import { getDirname, path } from 'vuepress/utils';
import { generateCatalogPage } from './generateCatalogPage.js';
import { catalogLocales as defaultLocales } from './locales.js';
import { PLUGIN_NAME, logger } from './logger.js';
const __dirname = getDirname(import.meta.url);
export const catalogPlugin = (options = {}) => (app) => {
    if (app.env.isDebug)
        logger.info('Options:', options);
    const { component, locales } = options;
    return {
        name: PLUGIN_NAME,
        define: () => ({
            __CATALOG_LOCALES__: getLocaleConfig({
                app,
                name: PLUGIN_NAME,
                default: defaultLocales,
                config: locales,
            }),
        }),
        onInitialized: () => generateCatalogPage(app, options),
        extendsBundlerOptions: (bundlerOptions) => {
            addViteSsrNoExternal(bundlerOptions, app, '@vuepress/helper');
        },
        ...(component
            ? {}
            : { clientConfigFile: path.join(__dirname, '../client/config.js') }),
    };
};
